node-backend-starter-code/public/index.html

```diff
@@ -0,0 +1,330 @@
+          function ajaxCall(method, apiUrl, data, isDataJSON, callback, params)
+          {
+              var request = new XMLHttpRequest();
+
+              request.open(method, apiUrl, true);
+              if (isDataJSON) {
+                  request.setRequestHeader('Content-Type', 'application/json');
+              }
+
+              request.onreadystatechange = function () {
+                  if (request.readyState != 4 || request.status != 200) {
+                      return;
+                  }
+
+                  try {
+                      result = JSON.parse(request.responseText);
+                  } catch (exception) {
+                      alert(apiErrorMsg);
+                      return false;
+                  }
+
+                  return callback(result, params);
+              };
+              request.send(data);
+          }
+
+          /**
+           * Helper function to escape single quotes in strings for use in Javascript
+           *
+           * @param  string text
+           * @return string
+           */
+          function escapeString(text)
+          {
+              return text.replace(/'/, "\\'");
+          }
+
+          /**
+           * Retrieve movie details for movie in current page of search results
+           *
+           * @param  int resultIndex
+           * @return false|object
+           */
+          function getMovie(resultIndex)
+          {
+              if (resultIndex < 0 || resultIndex >= searchResults.length) {
+                  return false;
+              }
+
+              return searchResults[resultIndex];
+          }
+
+          /**
+           * Post-AJAX callback when Favorites button is clicked
+           *
+           * @param  false|jsonObject result
+           * @param  object           params Callback params if any
+           * @return void
+           */
+          function favoritesCallback(result, params)
+          {
+              if (0 == result.length) {
+                  favoritesBox.innerHTML = 'You have not saved any movies to your favorites list yet.';
+                  return;
+              }
+
+              var html = '';
+
+              // List items
+              html += '<ol>';
+              result.forEach(function (element, index, array) {
+                  html += '<li>' + element.name + '</li>';
+              });
+              html += '</ol>';
+
+              favoritesBox.innerHTML = html;
+          }
+
+          /**
+           * Post-AJAX callback for findMovies to render search results
+           *
+           * @param  false|jsonObject result
+           * @param  object           params Callback params containing title and page
+           * @return void
+           */
+          function findMoviesCallback(result, params)
+          {
+              // Empty copy of previous search results
+              searchResults = [];
+
+              // No results found
+              if ("False" == result.Response) {
```

Comment:
Double quotes because it's coming from JSON? Seems to be a pattern you're using. Not disagreeing with it, just curious.

Also loose equality, why?